Add Kubernetes 1.18 Release Blog (#19826)
kubernetes/website: content/en/blog/_posts/2020-03-25-kubernetes-1.18-release-announcement.md

---
layout: blog

> title: "Kubernetes 1.18: Fit & Finish"

Reviewer: ```suggestion
title: 'Kubernetes 1.18: Fit & Finish'
```

Reviewer: There are issues in Netlify builds around double quotes and colon characters, but we've found single quotes fix that issue!

## Introducing Windows CSI support alpha for Kubernetes

With the release of Kubernetes 1.18, an alpha version of CSI Proxy for Windows is getting released. CSI proxy enables non-privileged (pre-approved) containers to perform privileged storage operations on Windows. CSI drivers can now be supported in Windows by leveraging CSI proxy.

## Other Updates

### Graduated to Stable 💯

- [Taint Based Eviction](https://github.com/kubernetes/enhancements/issues/166)

> - [Kubectl Diff](https://github.com/kubernetes/enhancements/issues/491)

Reviewer 2: ```suggestion
- [`kubectl diff`](https://github.com/kubernetes/enhancements/issues/491)
```

- [CSI Block storage support](https://github.com/kubernetes/enhancements/issues/565)

> - [APIServer DryRun](https://github.com/kubernetes/enhancements/issues/576)

Reviewer 2: ```suggestion
- [API Server dry run](https://github.com/kubernetes/enhancements/issues/576)
```

- [Pass Pod information in CSI calls](https://github.com/kubernetes/enhancements/issues/603)
- [Support Out-of-Tree vSphere Cloud Provider](https://github.com/kubernetes/enhancements/issues/670)
- [Support GMSA for Windows workloads](https://github.com/kubernetes/enhancements/issues/689)
- [Skip attach for non-attachable CSI volumes](https://github.com/kubernetes/enhancements/issues/770)

> - [Extend allowed PVC DataSources](https://github.com/kubernetes/enhancements/issues/989)

Reviewer 3: I would call this feature "PVC cloning"

Author: fixed

- [Moving kubectl package code to staging](https://github.com/kubernetes/enhancements/issues/1020)
- [RunAsUserName for Windows](https://github.com/kubernetes/enhancements/issues/1043)

> - [Adding AppProtocol to Services and Endpoints](https://github.com/kubernetes/enhancements/issues/1507)

Reviewer 2: ```suggestion
- [AppProtocol for Services and Endpoints](https://github.com/kubernetes/enhancements/issues/1507)
```

- [Extending Hugepage Feature](https://github.com/kubernetes/enhancements/issues/1539)
- [client-go signature refactor to standardize options and context handling](https://github.com/kubernetes/enhancements/issues/1601)

> - [Graduate NodeLocal DNSCache to GA ](https://github.com/kubernetes/enhancements/issues/1024)

Reviewer 2: ```suggestion
- [Node-local DNS cache](https://github.com/kubernetes/enhancements/issues/1024)
```

### Major Changes

- [EndpointSlice API](https://github.com/kubernetes/enhancements/issues/752)
- [Moving kubectl package code to staging](https://github.com/kubernetes/enhancements/issues/1020)

> - [Graduate Ingress to V1](https://github.com/kubernetes/enhancements/issues/1453)

Reviewer 2: This makes it look as if Ingress went GA, which I don't think is true.

Author: we're taking this out now

> - [Graduate Ingress to V1](https://github.com/kubernetes/enhancements/issues/1453)

Reviewer: ```suggestion
```

Reviewer: This can be removed, Ingress is going to GA in 1.19
